samples/sgs.py

```diff
@@ -0,0 +1,333 @@
+    return Ingredient(
+        name=ingredient_name,
+        text="".join(ingredient_lines),
+        simple_imports=simple_imports,
+        imports_from=imports_from,
+    )
+
+
+def load_ingredients(path: Path) -> dict:
+    ingredients = {}
+    for ipath in path.iterdir():
+        if ipath.is_dir():
+            ingredients.update(load_ingredients(ipath))
+        elif ipath.is_file():
+            ingredient = load_ingredient(ipath)
+            ingredients[ingredient.name] = ingredient
+    return ingredients
+
+
+def load_recipe(path: Path) -> str:
+    with path.open() as file:
+        return file.read()
+
+
+def load_recipes(path: Path) -> dict:
+    recipes = {}
+    for ipath in path.iterdir():
+        if ipath.is_dir():
+            recipes.update(load_recipes(ipath))
+        elif ipath.is_file():
+            recipes[ipath] = load_recipe(ipath)
+    return recipes
+
+
+def render_recipe(recipe: str, ingredients: dict) -> str:
+    """
+    Replace all `# IMPORTS` and `# INGREDIENT <name>` occurrences in
+    the provided recipe, producing a script ready to be saved to a file.
+    """
+    ingredients_used = []
+    file_lines = recipe.splitlines()
+
+    # Scan the file to used ingredients
+    for line in file_lines:
+        if match := INGREDIENT_FILL.match(line):
+            ingredients_used.append(ingredients[match.group(1)])
+
+    simple_imports_used = set()
+    for ingredient in ingredients_used:
+        for simple_import in ingredient.simple_imports:
+            simple_imports_used.add(simple_import)
+
+    from_imports_used = defaultdict(set)
+    for ingredient in ingredients_used:
+        for import_from in ingredient.imports_from:
+            from_imports_used[import_from[0]].add(import_from[1])
+
+    import_lines = set()
+    for simple_import in simple_imports_used:
+        if simple_import.asname:
+            import_lines.add(f"import {simple_import.name} as {simple_import.asname}")
+        else:
+            import_lines.add(f"import {simple_import.name}")
+
+    for module, from_imports in from_imports_used.items():
+        names = set()
+        for from_import in from_imports:
+            if from_import.asname:
+                name = f"{from_import.name} as {from_import.asname}"
+            else:
+                name = from_import.name
+            names.add(name)
+        names = ", ".join(names)
+        import_lines.add(f"from {module} import {names}")
+
+    import_lines = isort.code(
+        "\n".join(import_lines), config=isort.Config(profile="google")
+    )
+
+    output_file = []
+    header_added = False
+    for line in file_lines:
+        if IMPORTS_FILL.search(line):
+            output_file.append(import_lines)
+        elif match := INGREDIENT_FILL.search(line):
+            output_file.append(ingredients[match.group(1)].text)
+        elif REGION_START.search(line):
+            output_file.append(REGION_START.sub("# [START \\1]", line))
```

Comment:
We should not to forget to ignore region tag here. Are there any ways how to disable it (and in README also)?

Reply:
I honestly don't know... When I wrote this, I thought that region tag has to be at the beginning of the line. Now the snippet bot is complaining... I guess I'll need to do something silly and break it up so it doesn't match some regex.